internals/controllers/subnamespace/subnamespace_sync.go

```diff
@@ -0,0 +1,407 @@
+
+		var (
+			totalRequest, _ = allocated[res]
+			vRequest, _     = snsObject.Object.(*danav1.Subnamespace).Spec.ResourceQuotaSpec.Hard[res]
+		)
+		value := vRequest.Value() - totalRequest.Value()
+		freeToAllocate[res] = *resource.NewQuantity(value, resource.BinarySI)
+	}
+
+	return freeToAllocate
+}
+
+// deleteLegacyCRQ deletes a pre-exising CRQ in cases where they should only be a RQ.
+// This is needed because historically we used to only have CRQs and didn't introduce RQs until later
+func deleteLegacyCRQ(snsObject *utils.ObjectContext) error {
+	if utils.DoesSNSCrqExists(snsObject) {
+		crqName := snsObject.Object.GetName()
+		crqObj, err := utils.NewObjectContext(snsObject.Ctx, snsObject.Client, types.NamespacedName{Name: crqName}, &quotav1.ClusterResourceQuota{})
+		if err != nil {
+			return err
+		}
+		if err := crqObj.EnsureDeleteObject(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// syncSNSAnnotations sync subnamespace annotations
+func syncSNSAnnotations(snsObject, snsParentNS, parentSNS *utils.ObjectContext, isRq bool) error {
+	annots := snsObject.Object.GetAnnotations()
+
+	if isRq {
+		annots[danav1.IsRq] = danav1.True
+	} else {
+		annots[danav1.IsRq] = danav1.False
+	}
+
+	if upperRPName, isUpper, err := GetUpperResourcePoolName(snsObject, parentSNS); err != nil {
+		return err
+	} else {
+		annots[danav1.UpperRp] = upperRPName
+		annots[danav1.IsUpperRp] = isUpper
+	}
+
+	annots[danav1.CrqPointer] = utils.GetCrqPointer(snsObject.Object)
+
+	displayName := utils.GetNamespaceDisplayName(snsParentNS.Object) + "/" + snsObject.Object.GetName()
+	annots[danav1.OpenShiftDisplayName] = displayName
+	annots[danav1.DisplayName] = displayName
+
+	if err := snsObject.AppendAnnotations(annots); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// GetUpperResourcePoolName returns the name of the upper ResourcePool of a given subnamespace
+// and returns true if the SNS is the upper-rp and false otherwise
+func GetUpperResourcePoolName(sns *utils.ObjectContext, parentSNS *utils.ObjectContext) (string, string, error) {
+	isUpper := danav1.False
+	upperRPName := "none"
+
+	snsName := sns.Object.GetName()
+	snsParentName := parentSNS.GetName()
+
+	isUpperResourcePool, err := utils.IsUpperResourcePool(sns)
+	if err != nil {
+		return "", danav1.False, err
+	}
+
+	if isUpperResourcePool {
+		isUpper = danav1.True
+		upperRPName = snsName
+	}
+
+	isParentUpperResourcePool, ok := parentSNS.Object.GetAnnotations()[danav1.IsUpperRp]
+	if ok && isParentUpperResourcePool == danav1.True {
+		upperRPName = snsParentName
+	}
+
+	parentUpperRp, ok := parentSNS.Object.GetAnnotations()[danav1.UpperRp]
+	if ok && !isUpperResourcePool {
+		upperRPName = parentUpperRp
+	}
+
+	return upperRPName, isUpper, nil
+}
+
+// deleteNonUpperResourcePoolCRQ deletes the CRQ object corresponding to a non-upper ResourcePool
+func deleteNonUpperResourcePoolCRQ(snsObject *utils.ObjectContext) error {
+	quotaObj, err := utils.GetSNSQuotaObj(snsObject)
```

Comment:
im missing logs what you trying to delete dit you delete it succeffully?

Reply:
There are logs in the function that calls it